Headers/AppKit/NSTextCheckingClient.h

```diff
@@ -0,0 +1,113 @@
+/* Definition of class NSTextCheckingClient
+   Copyright (C) 2020 Free Software Foundation, Inc.
+   
+   By: Gregory John Casamento
+   Date: 02-08-2020
+
+   This file is part of the GNUstep Library.
+   
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#ifndef _NSTextCheckingClient_h_GNUSTEP_GUI_INCLUDE
+#define _NSTextCheckingClient_h_GNUSTEP_GUI_INCLUDE
+
+#import <Foundation/NSObject.h>
+#import <Foundation/NSRange.h>
+#import <Foundation/NSAttributedString.h>
+#import <Foundation/NSGeometry.h>
+
+#import <AppKit/NSTextInputClient.h>
+
+#if OS_API_VERSION(MAC_OS_X_VERSION_10_15, GS_API_LATEST)
+
+#if	defined(__cplusplus)
+extern "C" {
+#endif
+
+enum
+{
+ NSTextInputTraitTypeDefault,
+ NSTextInputTraitTypeNo,
+ NSTextInputTraitTypeYes
+};
+typedef NSInteger NSTextInputTraitType;
+
+@class NSDictionary, NSCandidateListTouchBarItem;
+
+@protocol NSTextInputTraits
+
+@optional
```

Comment:
Normally we use something like this for optional protocols

`#if GS_PROTOCOLS_HAVE_OPTIONAL
 @optional
#else
@end
@interface NSObject (NSFilePresenter)
#endif`

Not sure for which compilers this is still needed.

Reply:
It seems to work ok on clang and gcc in ci. Ill add those guards

Reply:
done.

Reply:
Somehow I am not seeing this change. What is wrong?

Reply:
Now I see it, sorry